SILInspector/AppDelegate.swift

```diff
@@ -176,39 +175,39 @@ class AppDelegate: NSObject, NSApplicationDelegate, NSTabViewDelegate {
 		asm = runProgram("swiftc - -emit-assembly")
 	}
 
-	func withParseAsLibrary(program:String) -> String {
-		return parseAsLibrary ? program + " -parse-as-library" : program
+	func withParseAsLibrary(_ program:String) -> String {
+		return parseAsLibrary ? program + " -parse-as-library -module-name SILInspector" : program
 	}
-	func withOptimize(program:String) -> String {
+	func withOptimize(_ program:String) -> String {
 		return optimize ? program + " -O" : program
 	}
-	func withModuleOptimize(program:String) -> String {
+	func withModuleOptimize(_ program:String) -> String {
 		return moduleOptimize ? program + " -whole-module-optimization" : program
 	}
-	func withDemangle(program:String) -> String {
+	func withDemangle(_ program:String) -> String {
 		return demangle ? program + " | xcrun swift-demangle" : program
 	}
 	
-	func runProgram(program:String) -> String {
+	func runProgram(_ program:String) -> String {
 		let toRun = withDemangle(withOptimize(withModuleOptimize(withParseAsLibrary(program))))
 		programText.stringValue = toRun
-		let inputPipe = NSPipe()
+		let inputPipe = Pipe()
 		let inputFile = inputPipe.fileHandleForWriting
-		inputFile.writeData(source.dataUsingEncoding(NSUTF8StringEncoding)!)
+		inputFile.write(source.data(using: .utf8)!)
 		inputFile.closeFile()
-		let task = NSTask()
+		let task = Process()
 		task.launchPath = "/bin/bash"
 		task.arguments = [ "-c", toRun]
-		let errorPipe = NSPipe()
-		let outputPipe = NSPipe()
+		let errorPipe = Pipe()
+		let outputPipe = Pipe()
 		task.standardInput = inputPipe
 		task.standardOutput = outputPipe
 		task.standardError = errorPipe
 		task.launch()
 		let outputData = outputPipe.fileHandleForReading.readDataToEndOfFile()
 		let errorData = errorPipe.fileHandleForReading.readDataToEndOfFile()
-		let output = NSString(data: outputData, encoding: NSUTF8StringEncoding) as! String
-		let error =  NSString(data: errorData, encoding: NSUTF8StringEncoding) as! String
+        let output = String(data: outputData, encoding: .utf8)!
```

Comment:
Indentation line on this looks a little off - can you fix? Looks like spaces instead of tabs (or vice versa ...)